src/main/java/core/org/akaza/openclinica/service/StudyEventServiceImpl.java

```diff
@@ -943,6 +943,51 @@ public void scheduleOrUpdateBulkEvent(MultipartFile file, Study study, String si
 
     }
 
+public void convertStudyEventStatus(String value, StudyEvent studyEvent){
+        StudyEventWorkflowStatusEnum workflowStatus=convertOriginalStudyEventStatusToWorkflowSatus(value);
+        if (workflowStatus!=null) {
+            studyEvent.setWorkflowStatus(workflowStatus);
+        }else{
+            if (value.equals("7")) {
+                studyEvent.setLocked(Boolean.TRUE);
+            }else if (value.equals("8")){
+                studyEvent.setSigned(Boolean.TRUE);
+            }
+        }
+}
+
+    public void convertStudyEventBeanStatus(String value, StudyEventBean studyEventBean){
+        StudyEventWorkflowStatusEnum workflowStatus=convertOriginalStudyEventStatusToWorkflowSatus(value);
+        if (workflowStatus!=null) {
+            studyEventBean.setWorkflowStatus(workflowStatus);
+        }else{
+            if (value.equals("7")) {
+                studyEventBean.setLocked(Boolean.TRUE);
+            }else if (value.equals("8")){
+                studyEventBean.setSigned(Boolean.TRUE);
+            }
+        }
+    }
+
+    public StudyEventWorkflowStatusEnum convertOriginalStudyEventStatusToWorkflowSatus(String value) {
```

Comment:
Instead of doing this conversion, is it possible to migrate the audit log data as well?

Reply:
I thought about migrating audit log but it will not be straight forward , since we are dividing study event 1 status type into 5 different status types. (Workflow, removed, archived, locked, signed). Migration time, we need to insert new records in auditlog table for each new status type. The same issue is with EventCrf statuses.

Reply:
Got it.